packages/page-staking/src/Performance/useCommitteePerformance.tsx

```diff
@@ -0,0 +1,272 @@
+}
+
+export function parseSessionBlockCount (sessionValidatorBlockCountValue: [StorageKey<AnyTuple>, Codec][], firstSessionBlockAuthor: string): [string, number][] {
+  return sessionValidatorBlockCountValue.map(([key, values]) => {
+    const account = key.args[0].toString();
+    let count = Number(values.toString());
+
+    if (account === firstSessionBlockAuthor) {
+      // a workaround for the fact that the first session block author is not reflected in that block
+      // elections.sessionValidatorBlockCount state
+      count += 1;
+    }
+
+    return [account, count];
+  });
+}
+
+function useSessionCommitteePerformanceImpl (sessions: number[]): SessionCommitteePerformance[] {
+  const { api } = useApi();
+
+  const MAX_POSSIBLE_SESSIONS = 1000;
+
+  const [firstBlockInSessionHashes, setFirstBlockInSessionHashes] = useState<(Hash | undefined)[]>(Array(MAX_POSSIBLE_SESSIONS).fill(undefined));
+  const [isPalletElectionsSupportedInSession, setIsPalletElectionsSupportedInSession] = useState<(boolean | undefined)[]>(Array(MAX_POSSIBLE_SESSIONS).fill(undefined));
+  const [lastBlockInSessionsHashes, setLastBlockInSessionsHashes] = useState<(Hash | undefined)[]>(Array(MAX_POSSIBLE_SESSIONS).fill(undefined));
+  const [firstSessionBlockAuthors, setFirstSessionBlockAuthors] = useState<(string | undefined)[]>(Array(MAX_POSSIBLE_SESSIONS).fill(undefined));
+  const [sessionValidatorBlockCountLookups, setSessionValidatorBlockCountLookups] = useState<[string, number][][]>(Array(MAX_POSSIBLE_SESSIONS).fill(undefined));
+  const [expectedValidatorBlockCountLookups, setExpectedSessionValidatorBlockCountLookups] = useState<[string, number][][]>(Array(MAX_POSSIBLE_SESSIONS).fill(undefined));
+  const [committees, setCommittees] = useState<string[][]>(Array(MAX_POSSIBLE_SESSIONS).fill(undefined));
+  // TODO comment why init is one off
+  const [committeeMemberPerformances, setCommitteeMemberPerformances] = useState<SessionCommitteePerformance[]>([]);
+
+  const MINIMUM_SUPPORTED_ELECTIONS_PALLET_VERSION = 3;
+
+  useEffect(() => {
+    if (api && api.consts.elections) {
+      const sessionPeriod = Number(api.consts.elections.sessionPeriod.toString());
+      const firstBlocksInSession = sessions.map((session) => session * sessionPeriod);
+
+      firstBlocksInSession.forEach((firstBlockInSession, index) => {
+        api.rpc.chain
+          .getBlockHash(firstBlockInSession)
+          .then((result): void => {
+            setFirstBlockInSessionHashes((existingItems) => {
+              return existingItems.map((item, j) => {
+                return j === index ? result : item;
+              });
+            });
+          })
+          .catch(console.error);
+      });
+    }
+  },
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  [api, JSON.stringify(sessions)]
+  );
+
+  useEffect(() => {
+    firstBlockInSessionHashes.forEach((firstBlockInSessionHash, index) => {
+      if (firstBlockInSessionHash) {
```

Comment:
nit: `if (!...) return;`

Reply:
This path is gone thankfully with the cleaner code.